plugin/src/test/java/org/opensearch/ml/action/custom_model/CustomModelITTests.java

```diff
@@ -0,0 +1,139 @@
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.transport.MLTaskResponse;
+import org.opensearch.ml.common.transport.unload.UnloadModelNodesResponse;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, numDataNodes = 1)
+public class CustomModelITTests extends MLCommonsIntegTestCase {
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    public void testCustomModelWorkflow() throws InterruptedException {
+        FunctionName functionName = FunctionName.TEXT_EMBEDDING;
+        String modelName = "all-MiniLM-L6-v2";
+        String version = "1.0.0";
+        MLModelFormat modelFormat = MLModelFormat.TORCH_SCRIPT;
+        String modelType = "bert";
+        TextEmbeddingModelConfig.FrameworkType frameworkType = TextEmbeddingModelConfig.FrameworkType.HUGGINGFACE_TRANSFORMERS;
+        int dimension = 384;
+        String allConfig = null;
+        String taskId = uploadModel(
+            functionName,
+            modelName,
+            version,
+            modelFormat,
+            modelType,
+            frameworkType,
+            dimension,
+            allConfig,
+            HUGGINGFACE_TRANSFORMER_MODEL_URL,
+            false
+        );
+        assertNotNull(taskId);
+
+        AtomicReference<String> modelId = new AtomicReference<>();
+        AtomicReference<MLModel> mlModel = new AtomicReference<>();
+        waitUntil(() -> {
+            String id = getTask(taskId).getModelId();
+            modelId.set(id);
+            MLModel model = null;
+            try {
+                if (id != null) {
+                    model = getModel(id + "_" + 1);
+                    mlModel.set(model);
+                }
+            } catch (Exception e) {
+                logger.error("Failed to get model " + id, e);
+            }
+            return id != null && model != null;
+        }, 20, TimeUnit.SECONDS);
+        assertNotNull(modelId.get());
+        MLModel model = getModel(modelId.get());
+        assertNotNull(model);
+        assertEquals(9, model.getTotalChunks().intValue());
+        assertNotNull(mlModel.get());
+        assertEquals(1, mlModel.get().getChunkNumber().intValue());
+
+        waitUntil(() -> {
+            SearchResponse response = searchModelChunks(modelId.get());
+            AtomicBoolean modelChunksReady = new AtomicBoolean(false);
+            if (response != null) {
+                long totalHits = response.getHits().getTotalHits().value;
+                if (totalHits == 9) {
+                    modelChunksReady.set(true);
+                }
+            }
+            return modelChunksReady.get();
+        }, 20, TimeUnit.SECONDS);
```

Comment:
Does this 20 seconds adds to the time needed to run integration tests? I see there are multiple 20 seconds waits here, so this means the tests would take very long?

Reply:
If you check the Github CI below, the build and test steps now takes about 5 minutes, latest 2.x Github [CI flow](https://github.com/opensearch-project/ml-commons/actions/runs/3322861550/jobs/5492471950) shows 4 minutes 10 seconds,